x-pack/plugins/security_solution/public/app/upgrade.ts

```diff
@@ -0,0 +1,176 @@
+  return `${name}-${version}`;
+};
+
+export const UpgradeEndpointPackage = () => {
+  const context = useKibana<StartServices>();
+  const { allEnabled: ingestEnabled } = useIngestEnabledCheck();
+
+  useEffect(() => {
+    const abortController = new AbortController();
+
+    // cancel any ongoing requests
+    const cleanup = () => {
+      abortController.abort();
+    };
+
+    if (ingestEnabled) {
+      const signal = abortController.signal;
+
+      (async () => {
+        try {
+          // make sure we're a privileged user before trying to install the package
+          const { success: hasPermissions } = await sendCheckPermissions(context.services.http, {
+            signal,
+          });
+
+          // if we're not a privileged user then return and don't try to check the status of the endpoint package
+          if (!hasPermissions) {
+            return cleanup;
+          }
+
+          // get the endpoint package's basic information
+          const endpointPackage = await sendGetEndpointSecurityPackage(context.services.http, {
+            signal,
+          });
+          if (endpointPackage.status === InstallationStatus.notInstalled) {
+            await sendInstallPackage(
+              context.services.http,
+              createPackageKey(endpointPackage.name, endpointPackage.version),
+              { signal }
+            );
+            return cleanup;
+          }
+
+          /**
+           * If a package is installed the `endpointPackage` variable will look like this:
+           * {
+           *  "name": "endpoint",
+           *  "version": "0.16.0-dev.1",
+           *  ...
+           *  "status": "installed",
+           *  "savedObject": {
+           *    ...
+           *    "attributes": {
+           *      ...
+           *      "name": "endpoint",
+           *      "version": "0.13.1",
+           *      "internal": false,
+           *      "removable": false,
+           *      "install_version": "0.13.1",
+           *      "install_status": "installed",
+           *      "install_started_at": "2020-09-15T19:52:25.153Z"
+           *    },
+           *    ...
+           *  }
+           *}
+           * So to get the actual installed version of the package we need to look in the `savedObject` section
+           */
+          const installedPackageKey = createPackageKey(
+            endpointPackage.savedObject.attributes.name,
+            endpointPackage.savedObject.attributes.version
+          );
+
+          // Just to be sure, lets get the full package info, which will include a `latestVersion` field for the endpoint package
+          const endpointPackageInfo = await sendGetPackageInfo(
+            context.services.http,
+            installedPackageKey,
+            { signal }
+          );
+
+          // check and see if the latest version is newer than the one we have installed
+          if (semver.gt(endpointPackageInfo.latestVersion, endpointPackageInfo.version)) {
```

Comment:
Is it ok to use `semver` here? The versions can be semver 2, so I figured it's the best way to do the comparison.

Reply:
I think including this is increasing the bundle size by quite a bit:
![image](https://user-images.githubusercontent.com/56361221/93362021-7f2dda80-f813-11ea-8add-92f3f5621a56.png)